osu.Game/Screens/Multiplayer/MultiUserPanel.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Screens.Multiplayer
+{
+    public class MultiUserPanel : ClickableContainer, IStateful<MultiUserPanel.UserState>
+    {
+        private UserState state;
+        public UserState State
+        {
+            get { return state; }
+            set
+            {
+                if (state == value)
+                    return;
+
+                state = value;
+                switch (state)
+                {
+                    case UserState.Guest:
+                        statusString = "Hosting a Room";
+                        UpdatePanel(this);
+                        break;
+
+                    case UserState.Host:
+                        statusString = "Waiting in a Room";
+                        UpdatePanel(this);
+                        break;
+                }
+            }
+        }
+
+        private string userName;
+        private string statusString;
+        private OsuSpriteText userSprite;
+        private OsuSpriteText statusSprite;
+
+        public const int PANEL_HEIGHT = 100;
+        public const int STATUS_HEIGHT = 33;
+        public const float PANEL_WIDTH = 0.25f;
+
+        public void UpdatePanel(MultiUserPanel panel)
```

Comment:
Is this supposed to do more than just set the spritetext? otherwise just make ```statusString``` a property that sets the spritetext